token/cli/src/command.rs

```diff
@@ -1549,28 +1557,37 @@ async fn command_burn(
     config: &Config<'_>,
     account: Pubkey,
     owner: Pubkey,
-    ui_amount: f64,
+    ui_amount: Option<f64>,
     mint_address: Option<Pubkey>,
     mint_decimals: Option<u8>,
     use_unchecked_instruction: bool,
     memo: Option<String>,
     bulk_signers: BulkSigners,
 ) -> CommandResult {
-    println_display(
-        config,
-        format!("Burn {} tokens\n  Source: {}", ui_amount, account),
-    );
-
     let mint_address = config.check_account(&account, mint_address).await?;
     let mint_info = config.get_mint_info(&mint_address, mint_decimals).await?;
-    let amount = spl_token::ui_amount_to_amount(ui_amount, mint_info.decimals);
     let decimals = if use_unchecked_instruction {
         None
     } else {
         Some(mint_info.decimals)
     };
 
     let token = token_client_from_config(config, &mint_info.address, decimals)?;
+
+    let balance = token.get_account_info(&account).await?.base.amount;
+    let amount = ui_amount
+        .map(|ui_amount| spl_token::ui_amount_to_amount(ui_amount, mint_info.decimals))
+        .unwrap_or(balance);
```

Comment:
This will break offline signing, since it'll always try to fetch an account. Can you only fetch this balance if `ui_amount` is `None` AND we're not in sign-only mode?

I couldn't come up with a neat way to do this because of the async function, so I'll accept whatever you come up with, but here's my example:

```suggestion
    let amount = if let Some(ui_amount) = ui_amount {
        spl_token::ui_amount_to_amount(ui_amount, mint_info.decimals)
    } else {
        if config.sign_only {
            return Err("Burning all tokens requires online signing").into();
        }
        token.get_account_info(&account).await?.base.amount
    };
```

Reply:
Fixed with https://github.com/solana-labs/solana-program-library/pull/6057/commits/50a0f0606510feb92c0244d75943ee8597b9a561